app/scripts/views/settings/account_recovery/recovery_key.js

```diff
@@ -0,0 +1,98 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import $ from 'jquery';
+import Cocktail from 'cocktail';
+import BaseView from '../../base';
+import ModalSettingsPanelMixin from '../../mixins/modal-settings-panel-mixin';
+import Template from 'templates/settings/account_recovery/recovery_key.mustache';
+import RecoveryKeyPrintTemplate from 'templates/settings/account_recovery/recovery_key_print.mustache';
+
+const {t} = BaseView;
+
+const View = BaseView.extend({
+  template: Template,
+  className: 'account-recovery-key',
+  viewName: 'settings.account-recovery.recovery-key',
+
+  events: {
+    'click .done-link': '_done',
+    'click .download-key': '_downloadKey',
+    'click .print-key': '_printKey'
+  },
+
+  _formatRecoveryKey(key) {
+    if (key) {
+      // Insert spaces every 4 characters and remove trailing space
+      return key.replace(/(\w{4})/g, '$1 ').replace(/(^\s+|\s+$)/, '');
+    }
+  },
+
+  _getFormatedRecoveryKeyFilename() {
+    const account = this.getSignedInAccount();
+    let formattedFilename = account.get('email') + ' ' + t('Firefox Recovery Key');
```

Comment:
Can filenames contain emoji characters? Email addresses can if created programatically, though we do not yet allow that on the front end.

Reply:
Tested on OSX and filenames can have emoji characters. I think this might be ok for now.